tests/wpt/servowpt.py

```diff
@@ -186,41 +188,106 @@ def update_tests(**kwargs):
     return 1 if return_value is update.exit_unclean else 0
 
 
-class TrackerFilter():
+class GithubContextInformation(NamedTuple):
+    build_url: str
+    pull_url: str
+    branch_name: str
+
+
+class TrackerDashboardFilter():
     def __init__(self):
-        self.url = os.environ.get(TRACKER_API_ENV_VAR, TRACKER_API)
-        if self.url.endswith("/"):
-            self.url = self.url[0:-1]
+        base_url = os.environ.get(TRACKER_API_ENV_VAR, TRACKER_API)
+        self.headers = {
+            "Content-Type": "application/json"
+        }
+        if TRACKER_DASHBOARD_SECRET_ENV_VAR in os.environ:
+            self.url = f"{base_url}/dashboard/attempts"
+            secret = os.environ[TRACKER_DASHBOARD_SECRET_ENV_VAR]
+            self.headers["Authorization"] = f"Bearer {secret}"
+        else:
+            self.url = f"{base_url}/dashboard/query"
 
-    def is_failure_intermittent(self, test_name):
-        query = urllib.parse.quote(test_name, safe='')
-        request = urllib.request.Request("%s/query.py?name=%s" % (self.url, query))
-        search = urllib.request.urlopen(request)
-        return len(json.load(search)) > 0
+    @staticmethod
+    def get_github_context_information() -> GithubContextInformation:
+        github_context = json.loads(os.environ.get("GITHUB_CONTEXT", "{}"))
+        if not github_context:
+            return GithubContextInformation("", "", "")
```

Comment:
wdyt about using `Optional[str]` and `None` here, rather than `""`? (the dashboard api accepts `null` for these, so no changes to the request body would be needed)

Reply:
That makes total sense to me. I wasn't sure if the server supported that. I've made this change.